spec/predicates/runtime-trace.md

```diff
@@ -0,0 +1,2677 @@
+      "process": [
+         { /* object */ }
+      ],
+      "fileAccess": [
+        { /* object */ }
+      ],
+    },
+    "metadata": {
+      "buildStartedOn": "<TIMESTAMP>",
+      "buildFinishedOn": "<TIMESTAMP>",
+    }
+  }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the in-toto attestation parsing rules. Summary:
+
+-   Consumers MUST ignore unrecognized fields.
+-   The `predicateType` URI includes the major version number and will always change whenever there is a backwards incompatible change.
+-   Minor version changes are always backwards compatible and “monotonic.” Such changes do not update the `predicateType`.
+-   Producers MAY add extension fields using field names that are URIs.
+-   _optional_ fields MAY be unset or null, and should be treated equivalently. Both are equivalent to empty for object or array values.
+
+## Fields
+
+`monitor` _object_, _required_
+
+Identifies the specific monitor instance used to trace the runtime.
+
+`monitor.id` _string (URI)_, _required_
+
+URI indicating the monitor’s identity.
+
+`build` _object_, _required_
+
+Feedback required: Should this predicate be scoped to runtime traces of builds only or generalize to runtime traces of any activity?
+
+`builder.builderId` _string (URI)_, _required_
+
+URI indicating the builder’s identity. Ex: GitLab Runner ID
+
+`build.type` _string (TypeURI)_, _required_
+
+URI indicating the type of build performed.
+
+`build.event` _string_, _required_
+
+String identifying the specific job or task associated with the attestation.
+
+`invocation` _object_, _optional_
+
+Identifies the event that initiated the monitor. Similar to `invocation` in SLSA Provenance.
+
+`invocation.configSource` _object_, _optional_
+
+Effectively a pointer to the monitor's configuration.
+
+`invocation.tracePolicy` _object_, _optional_
+
+Feedback required: Should tracingPolicy point outwards to a separate attestation with the same subject that wraps around the actual policy? The policy can be more securely defined and distributed (signed for by developer keys rather than online keys for example).
+
+`monitorLog` _object_, _required_
+
+Feedback required: Should different types of monitors (process vs network for example) emit different attestation subtypes? If they are all combined in one, should there still be some differentiation within `monitorLog`?
+
+`monitorLog.process` _list_, _optional_
+
+`monitorLog.network` _list_, _optional_
+
+`monitorLog.fileAccess` _list_, _optional_
+
+Record of files accessed during the build process. A complete list of _materials_ can be derived from this information. Each entry in this list is expected to record the path of the file and one or more digests of the file. This field is a list rather than a key value map because a single file may be used multiple times during the build process. Further, some files that are accessed may _change_ during the build process, and so, different entries may have different digests.
```

Comment:
every other monitorLog (process, network) was abstract, this is the only one that specifies the contents of the trace. I think at this stage we should keep all of them abstract.

Reply:
In order to create policy over the attestation we need to have a defined schema for the monitor log.

Reply:
oh I see. policy use case was not mentioned in this spec so I presumed the use case was to enrich provenance data. 
I have a feeling that discussing the detailed logs schema will be much longer then the outline schema. If that's the intention, let's see the schemas and then we can discuss.